app/javascript/components/DecisionTree.jsx

```diff
@@ -0,0 +1,290 @@
+                    [id]: { id, message: '', options: [] },
+                    [currentId]: {
+                      ...currentNode,
+                      options: [
+                        ...currentNode.options,
+                        { label: '', next: id }
+                      ]
+                    }
+                  }})
+                this.setState({
+                  path: [
+                    ...this.state.path.slice(0, pathIx + 1),
+                    id
+                  ],
+                  editingOptionTo: id
+                })
+              }}
+              updateOption={(ix, value) => {
+                onChange({initial, nodes: {
+                  ...nodes,
+                  [currentId]: {
+                    ...currentNode,
+                    options: [
+                      ...currentNode.options.slice(0, ix),
+                      { ...currentNode.options[ix], label: value },
+                      ...currentNode.options.slice(ix + 1)
+                    ]
+                  }
+                }})
+              }}
+              selectOption={(ix) => {
+                this.setState({
+                  path: [
+                    ...this.state.path.slice(0, pathIx + 1),
+                    ...this._buildDefaultPath(nodes, currentNode.options[ix].next)
+                  ]
+                })
+              }}
+              deleteOption={(ix) => {
+                onChange({initial, nodes: this._deleteOption(nodes, currentNode, ix)})
+              }}
+              focusedOption={(ix) => {
+                this.setState({
+                  ...this.state,
+                  editingOptionTo: null
+                })
+              }}
+            />
+          )
+        })}
+      </div>
+    )
+  }
+}
+
+class TreeNode extends Component {
+
+  render() {
+    const { node, nextNodeId, editingOptionTo, updateMessage,
+      addOption, updateOption, selectOption, deleteOption, focusedOption } = this.props
+
+    return (
+      <Paper
+        className="tree-node-container"
+        zDepth={1} >
+        <Field
+          id={`tree-node-message-${node.id}`}
+          className="tree-node-message"
+          placeholder="Message"
+          value={node.message}
+          onChange={updateMessage} />
+        <ul className="invisible-scrollbar">
+          {
+            node.options.map((option, ix) => (
+              <li key={ix}>
+                <TreeNodeOption
+                  id={ix}
+                  selected={nextNodeId == option.next}
+                  focused={editingOptionTo == option.next}
+                  option={option}
+                  onChange={(value) => updateOption(ix, value)}
+                  onSelect={() => selectOption(ix)}
+                  onDelete={() => deleteOption(ix)}
+                  onFocused={(ix) => focusedOption(ix)} />
+              </li>
+            ))
+          }
+        </ul>
+        <div className="option-toolbar">
```

Comment:
Please, add a pending ui in tech debt to Maybe include the add option in the flow of the options